scipy/spatial/_spherical_voronoi.py

```diff
@@ -311,3 +301,38 @@ def calculate_areas(self):
 
         # Get polygon areas using A = omega * r**2
         return solid_angles * self.radius**2
+
+    def _calculate_areas_2d(self):
+        # Find the start and end points of each circular arc
+        arcs = self.points[self._simplices] - self.center
+
+        # Calculate the angle subtended by each arc
+        cosine = np.einsum('ij,ij->i', arcs[:, 0], arcs[:, 1])
+        sine = np.abs(np.linalg.det(arcs))
+        theta = np.arctan2(sine, cosine)
+
+        # Get areas using A = r * theta
+        return self.radius * theta
+
+    def calculate_areas(self):
+        """Calculates the areas of the Voronoi regions.
+
+        For 2D point sets, the regions are circular arcs. The sum of the areas
+        is `2 * pi * radius`.
+
+        For 3D point sets, the regions are spherical polygons. The sum of the
+        areas is `4 * pi * radius**2`.
+
+        .. versionadded:: 1.5.0
+
+        Returns
+        -------
+        areas : double array of shape (npoints,)
+            The areas of the Voronoi regions.
+        """
+        if self._dim == 2:
+            return self._calculate_areas_2d()
+        elif self._dim == 3:
+            return self._calculate_areas_3d()
+        else:
+            raise TypeError("Only supported for 2D and 3D point sets")
```

Comment:
Might as well add a simple test for the raising of this exception

Reply:
Added.